In [2]:
import pandas as pd
df=pd.read_csv("Sample - Superstore.csv",encoding="latin1")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [3]:
print(df.shape)
df.info()
df.describe()

(4578, 21)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4578 entries, 0 to 4577
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         4578 non-null   int64  
 1   Order ID       4578 non-null   object 
 2   Order Date     4578 non-null   object 
 3   Ship Date      4578 non-null   object 
 4   Ship Mode      4578 non-null   object 
 5   Customer ID    4578 non-null   object 
 6   Customer Name  4578 non-null   object 
 7   Segment        4578 non-null   object 
 8   Country        4578 non-null   object 
 9   City           4578 non-null   object 
 10  State          4578 non-null   object 
 11  Postal Code    4578 non-null   int64  
 12  Region         4578 non-null   object 
 13  Product ID     4578 non-null   object 
 14  Category       4578 non-null   object 
 15  Sub-Category   4578 non-null   object 
 16  Product Name   4578 non-null   object 
 17  Sales          4578 non-null   float64
 1

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,4578.000000,4578.000000,4578.000000,4578.000000,4578.000000,4578.000000
mean,2289.500000,54879.289646,232.425959,3.802315,0.156686,26.212476
std,1321.699096,31767.037835,650.125232,2.245579,0.206231,211.017604
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-3839.990400
25%,1145.250000,22920.750000,16.768000,2.000000,0.000000,1.747800
50%,2289.500000,55122.000000,52.728000,3.000000,0.200000,8.713900
75%,3433.750000,90004.000000,213.394000,5.000000,0.200000,28.847950
max,4578.000000,99207.000000,22638.480000,14.000000,0.800000,5039.985600


1.Which product caegory/sub-category generates the most sales and profit?
2.Which region or state performs best/worst in profit?
3.Are there orders sold at a loss-and what do they have in common?
4.How does discoubnt level relate to profit?
5.Is ther a seasonal trend in sales over time?


In [5]:
category_sales = df.groupby("Category")["Sales"].sum().sort_values(ascending=False)
print(category_sales)

category_profit = df.groupby("Category")["Profit"].sum().sort_values(ascending=False)
print(category_profit)

Category
Technology         406939.4570
Furniture          330574.6899
Office Supplies    326531.8950
Name: Sales, dtype: float64
Category
Technology         58012.3697
Office Supplies    54078.2733
Furniture           7910.0708
Name: Profit, dtype: float64


In [7]:
region_sales = df.groupby("Region")["Sales"].sum().sort_values(ascending=False)
print(region_sales)

region_profit = df.groupby("Region")["Profit"].sum().sort_values(ascending=False)
print(region_profit)

Region
West       329684.4745
East       306833.5790
Central    230987.6244
South      196540.3640
Name: Sales, dtype: float64
Region
West       43135.1023
East       40471.4019
South      21377.3504
Central    15016.8592
Name: Profit, dtype: float64


In [8]:
loss_orders = df[df["Profit"]<0]
print("Number of loss-making orders:",loss_orders.shape[0])

print(loss_orders["Category"].value_counts())
print(loss_orders["Sub-Category"].value_counts())
print(loss_orders["Discount"].describe())

Number of loss-making orders: 834
Category
Office Supplies    411
Furniture          310
Technology         113
Name: count, dtype: int64
Sub-Category
Binders        292
Chairs          98
Tables          97
Storage         72
Furnishings     70
Phones          57
Bookcases       45
Accessories     34
Appliances      26
Machines        22
Supplies        17
Fasteners        4
Name: count, dtype: int64
count    834.000000
mean       0.487542
std        0.234813
min        0.150000
25%        0.200000
50%        0.500000
75%        0.700000
max        0.800000
Name: Discount, dtype: float64


In [9]:
correlation = df["Discount"].corr(df["Profit"])
print("Correlation between Discount and Profit:",correlation)
df.groupby("Discount")["Profit"].mean().sort_index()

Correlation between Discount and Profit: -0.24133040544580842


,Profit
Discount,
0.00,64.543423
0.10,108.666543
0.15,25.661012
0.20,23.060056
0.30,-45.029055
0.32,-90.686769
0.40,-108.516700
0.45,-194.736200
0.50,-468.457477


In [10]:
df["Order Date"] = pd.to_datetime(df["Order Date"])
df["Order Month"] = df["Order Date"].dt.to_period("M")

monthly_sales = df.groupby("Order Month")["Sales"].sum()
print(monthly_sales)

Order Month
2014-01     6221.2980
2014-02     1362.5060
2014-03    33887.9310
2014-04    13391.8550
2014-05    11089.2350
2014-06    24693.5356
2014-07    21470.4440
2014-08    15682.8285
2014-09    46543.1318
2014-10    15259.9610
2014-11    41441.7980
2014-12    35121.7485
2015-01     6350.7880
2015-02     4390.1950
2015-03    21112.0690
2015-04    15141.1470
2015-05     6592.7280
2015-06     8496.5730
2015-07    22436.9230
2015-08    22149.9320
2015-09    30294.9180
2015-10    15318.3665
2015-11    26074.4503
2015-12    35779.2892
2016-01     5824.5750
2016-02     5880.8100
2016-03    22852.1520
2016-04    23422.7920
2016-05    25993.0300
2016-06    15271.6600
2016-07    21959.7630
2016-08    14315.7518
2016-09    32821.5979
2016-10    12453.1440
2016-11    28481.3858
2016-12    34062.3790
2017-01    16196.8670
2017-02     7832.3420
2017-03    17403.4808
2017-04    24015.7821
2017-05    20924.1970
2017-06    19481.5225
2017-07    17736.4750
2017-08    25180.1260
2017-09    39853.824

*Dataset overview:9994 orders 21 columns,no missing values,covering Sales/Profit/Discount across Category,Region and time.  

*Data isue found: 834 orders at 8% are sold at a loss, concentrated in office supplies_especiallybinders.

*Seasonal trend: Sales consistently dip in jan/feb and peak in sep,nov,and dec each year - likely holiday-driven demand.



